src/pytest_memray/marks.py

```diff
@@ -1,34 +1,65 @@
 from __future__ import annotations
 
+from dataclasses import dataclass
+from typing import Optional
+from typing import Tuple
+
 from memray import AllocationRecord
 
 from .utils import parse_memory_string
 from .utils import sizeof_fmt
 
+PytestSection = Tuple[str, str]
+
+
+# REVIEW bikeshed the name?
+@dataclass
+class FailedTestMemoryInfo:
+    """Type that holds all memray-related info for a failed test."""
+
+    max_memory: float
+    total_allocated_memory: int
+    allocations: list[AllocationRecord]
+
+    @property
+    def section(self) -> PytestSection:
+        """Return a tuple in the format expected by section reporters."""
+        total_memory_str = sizeof_fmt(self.total_allocated_memory)
+        max_memory_str = sizeof_fmt(self.max_memory)
+        text_lines = [
+            f"Test is using {total_memory_str} out of limit of {max_memory_str}",
+            "List of allocations: ",
+        ]
+        for record in self.allocations:
+            size = record.size
+            stack_trace = record.stack_trace()
+            if not stack_trace:
+                continue
+            (function, file, line), *_ = stack_trace
+            text_lines.append(f"\t- {function}:{file}:{line} -> {sizeof_fmt(size)}")
+
+        return "memray-max-memory", "\n".join(text_lines)
+
+    @property
+    def long_repr(self) -> str:
+        """Generate a longrepr user-facing error message."""
+        total_memory_str = sizeof_fmt(self.total_allocated_memory)
+        max_memory_str = sizeof_fmt(self.max_memory)
+        return (
+            f"""Test was limited to {max_memory_str} but allocated {total_memory_str}"""
```

Comment:
Remove triple quote, single suffices; also can we shorten the variable names to fit this return into a single line?